scrapy/contrib/pipeline/images.py

```diff
@@ -90,11 +90,25 @@ def get_images(self, response, request, info):
             yield thumb_path, thumb_image, thumb_buf
 
     def convert_image(self, image, size=None):
-        if image.format == 'PNG' and image.mode == 'RGBA':
-            background = Image.new('RGBA', image.size, (255, 255, 255))
-            background.paste(image, image)
-            image = background.convert('RGB')
-        elif image.mode != 'RGB':
+        if image.format == 'PNG':
+            # try to add background to all .png images
+            # regardless of their mode
+            if image.mode != 'RGBA':
+                try:
+                    # we can only add background if mode is RGBA
+                    # so convert to this mode
+                    image = image.convert('RGBA')
+                    add_background = True
+                except ValueError:
+                    add_background = False
+            if add_background:
```

Comment:
Isn't this wrong? If background is converted `add_background` will be `True`, but the comment says `# if image is not RGBA adding background will fail`